### ISSUE TITLE
problems with object with properties in utf8 encoding

### ISSUE THREAD

**Issue by pdenapo** [oven-sh/bun#2928]
### What version of Bun is running?

0.6.1

### What platform is your computer?

Linux 5.10.0-16-amd64 x86_64 unknown (Debian 11)

### What steps can reproduce the bug?

Consider the following javascript code

```javascript
let object1 = {}
object1["código"] = 12
console.log(object1)

let object2 = {}
object2.código = 12
console.log(object2)

console.log("código=", object2["código"])
```

### What is the expected behavior?

In bun (version 0.6.1) the output is

{
  "código": 12
}
{
  "cÃ³digo": 12
}
código= undefined

### What do you see instead?

In node.js (version 19.9.0) the output is

{ 'código': 12 }
{ 'código': 12 }
código= 12

### Additional information

I think that writing object2.código or object2["código"] should be equivalent in your code.

As you can see this can break code written for node.js where variable names in utf8 are used.
The bug I guess is at the parsing level.

Also the ouput of console.log in an object is different in bun.js  is different for object1
(more spacing: three lines instead of one). But I'm not sure if this should be considered
a bug, as bun output is nicer!

**Comment by Jarred-Sumner** [comment]
yeah this is a bug and i'm surprised we lack test coverage for this

**Comment by pdenapo** [comment]
"What is the expected behavior" and "What do you see instead" are interchanged! LOL

**Comment by pdenapo** [comment]
I confirm that is fixed in Bun 0.6.3 (canary). Thank you!
